plugins/modules/opkg.py

```diff
@@ -154,13 +183,20 @@ def install_packages(module, opkg_path, packages):
     install_c = 0
 
     for package in packages:
-        if query_package(module, opkg_path, package):
+        package, version = split_name_and_version(module, package)
+
+        if query_package(module, opkg_path, package, version):
             continue
 
-        rc, out, err = module.run_command("%s install %s %s" % (opkg_path, force, package))
+        if version is not None:
+            version_str = "=%s" % version
+        else:
+            version_str = ""
 
-        if not query_package(module, opkg_path, package):
-            module.fail_json(msg="failed to install %s: %s" % (package, out))
+        rc, out, err = module.run_command("%s install %s %s%s" % (opkg_path, force, package, version_str))
```

Comment:
Better use a list here as well.

Reply:
Changed in [beaf2a1](https://github.com/ansible-collections/community.general/pull/5688/commits/beaf2a186bed1eda88a8f7c1243dc2be4aba6290)